Classes/Capture/AddNoteViewController.swift

```diff
@@ -54,12 +54,114 @@ final class AddNoteViewController: UIViewController {
         fatalError("init(coder:) has not been implemented")
     }
 
+    private lazy var toolsBar: UIToolbar = {
+        let bar = UIToolbar()
+        let button1 = UIButton( type: .custom )
+        button1.setImage( UIImage( named: "linkkey.png" ), for: .normal )
+        button1.addTarget(self, action: #selector(addLinkMarkup), for: .touchUpInside)
+        button1.frame = CGRect( x: 0, y: 0, width: 53, height: 51 )
+        button1.bounds = CGRect( x: 0, y: 0, width: 53, height: 51 )
+        let urlButton = UIBarButtonItem( customView: button1 )
+        let button2 = UIButton( type: .custom )
+        button2.setImage( UIImage( named: "datepick.png" ), for: .normal )
+        button2.addTarget(self, action: #selector(setDatePickerView), for: .touchUpInside)
+        button2.frame = CGRect( x: 0, y: 0, width: 53, height: 51 )
+        button2.bounds = CGRect( x: 0, y: 0, width: 53, height: 51 )
+        let dateButton = UIBarButtonItem( customView: button2 )
+        let spacer = UIBarButtonItem(barButtonSystemItem: .flexibleSpace, target: nil, action: nil)
+        bar.items = [urlButton, dateButton, spacer ]
+        bar.sizeToFit()
+        return bar
+    }()
+
+    @objc func addLinkMarkup() {
+        guard self.textView.text != nil else { return } // If no selected text, ignore.  Consider popup?
+        if let range = self.textView.selectedTextRange, !range.isEmpty {
+            let selectedText = self.textView.text( in: range )
+            self.textView.replace( range, withText: "[[\(selectedText!)][]]" )
+            let newCursor = self.textView.position(from: range.end, offset: 4)!
+            self.textView.selectedTextRange = self.textView.textRange( from: newCursor, to: newCursor)
+        }
+    }
+
+    let datePicker: UIDatePicker = {
+        let picker = UIDatePicker(frame: CGRect(x: 0, y: 0, width: UIScreen.main.bounds.width, height: 216))
+        picker.datePickerMode = .dateAndTime
+        return picker
+    }()
+
+    private lazy var datePickBar: UIToolbar = {
+        let bar = UIToolbar()
+        let cancelButton = UIBarButtonItem( title: "Cancel", style: .plain, target: self, action: #selector(setDefaultInputView) )
+        let scheduleButton = UIBarButtonItem( title: "Schd", style: .plain, target: self, action: #selector(insertDateSchedule) )
+        let deadlineButton = UIBarButtonItem( title: "Dead", style: .plain, target: self, action: #selector(insertDateDeadline) )
```

Comment:
Not sure about these abbreviations, I understand that they will not fit otherwise but still. @webframp, what do you think?

Reply:
I don't have any good suggestions for alternatives but agree it doesn't quite fit for me. I think I'm probably be fine with them unless we have suggestions for clearer alternatives.